engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -294,19 +297,44 @@ - (void)sharedSetupWithProject:(nullable FlutterDartProject*)project
                                                 opaque:_viewOpaque
                                        enableWideGamut:engine.project.isWideGamutEnabled];
   [_engine createShell:nil libraryURI:nil initialRoute:initialRoute];
+  [self performImplicitEngineCallbacks];  // This is the earliest this can be called because it
```

Comment:
I missed the context here before; since this is an `init`, it's dangerous to call `self`. Flutter engine code is terrible about this (see the TODO a few lines below), but we shouldn't make it worse. It would be better to either inline all that code, or make it a static C function, to avoid accidental obfuscated calls to `self`.

Reply:
Argh, I just discovered that I left this comment in draft so I didn't go out yesterday.